File: dav.client/ClientApp/src/app/models/order.ts
import { ProductLine } from "./ProductLine";
import { OrderStatus } from "./order-status.enum";

export class Order {
  public id: number;
  public supplierId: number;
  public supplerName: string;
  public orderNumber: number;
  public dateTime: Date;
  public statusId: OrderStatus;
  public productLines: ProductLine[];

  constructor(){
    this.productLines = [new ProductLine()];
  }
}

File: DAV.Domain/Entities/Order.cs
using DAV.Domain.Interfaces;
using DAV.Domain.Utilities;
using System;
using System.Collections;
using System.Collections.Generic;

namespace DAV.Domain.Entities
{
    public class Order : IEntity, IEntityLookup
    {
        public int? Id { get; set; }

        public int SupplierId { get; set; }

        public DateTime DateTime { get; set; }

        public OrderStatus StatusId { get; set; }

        public string Status => StatusId.ToString();

        public bool? IsActive { get; set; } = true;

        public object LookupId => Id;

        public object LookupValue => Id;

        public IEnumerable<ProductLine> ProductLines { get; set; } = new List<ProductLine>();
    }
}
